src/main/java/it/gov/pagopa/pu/sil/registry/RegistryLogger.java

```diff
@@ -0,0 +1,130 @@
+package it.gov.pagopa.pu.sil.registry;
+
+import it.gov.pagopa.pu.auth.dto.generated.UserInfo;
+import it.gov.pagopa.pu.sil.enums.RegistryEventSubType;
+import it.gov.pagopa.pu.sil.enums.RegistrySilEventType;
+import it.gov.pagopa.pu.sil.enums.SilOutcome;
+import it.gov.pagopa.pu.sil.event.producer.RegistryProducerService;
+import it.gov.pagopa.pu.sil.service.soap.JAXBTransformService;
+import it.gov.pagopa.pu.sil.util.Utilities;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.stereotype.Service;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+@Service
+@Slf4j
+public class RegistryLogger {
+
+  public static final String SKIP_XML_BODY_KEY = "skipXmlBody";
+  public static final String XML_BODY_KEY = "xmlBody";
+
+  private final JAXBTransformService jaxbTransformService;
+
+  private final RegistryProducerService registryProducerService;
+
+  public RegistryLogger(JAXBTransformService jaxbTransformService, RegistryProducerService registryProducerService) {
+    this.jaxbTransformService = jaxbTransformService;
+    this.registryProducerService = registryProducerService;
+  }
+
+  public <I, O> O execute(String orgFiscalCode, RegistrySilEventType eventType, String iuv, I request, UserInfo loggedUser, String serviceSilId,
+                          Supplier<Pair<O, SilOutcome>> requestHandler, Function<Exception, O> exceptionHandler) {
+    return this.execute(orgFiscalCode, eventType, iuv, request, loggedUser, serviceSilId, requestHandler, exceptionHandler, null, null);
+  }
+
+  public <I, O> O execute(String orgFiscalCode, RegistrySilEventType eventType, String iuv, I request, UserInfo loggedUser, String serviceSilId,
+                          Supplier<Pair<O, SilOutcome>> requestHandler, Function<Exception, O> exceptionHandler,
+                          Supplier<Map<String, Object>> registryBodyRequestExtraInfoRetriever, Function<O, Map<String, Object>> registryBodyResponseExtraInfoExtractor) {
+    produceReqRegistryEvent(orgFiscalCode, eventType, iuv, request, loggedUser, serviceSilId, registryBodyRequestExtraInfoRetriever);
+    Pair<O, SilOutcome> response2outcome = Pair.of(null, SilOutcome.KO);
```

Comment:
not needed, it will always be initialized on try or catch
```suggestion
    Pair<O, SilOutcome> response2outcome;
```

Reply:
if I remove the initialization, a compile error occurs 6 lines below:
Variable 'response2outcome' might not have been initialized